src/time/forward.h

```diff
@@ -25,8 +26,21 @@ extern struct timeval current_time;
 extern double current_dtime;
 extern time_t squid_curtime;
 
-time_t getCurrentTime(void);
-int tvSubMsec(struct timeval, struct timeval);
+/// Update squid_curtime, current_time, and current_dtime
+/// \returns new value of squid_curtime
+time_t getCurrentTime();
```

Comment:
Alex, 21 Dec 2021:
> We (reluctantly) use "gadgets" when we cannot find a common word for the collection of assorted APIs.

The API in this case is ```namespace Time```. Looking towards replacing all these definitions with ```<chrono>``` API calls soon. IMO it is not worth adding a gadgets file, just to remove it in the followups to libtime.

If you insist on not using "forward.h" I will rename this to "old_api.h"

Reply:
I picked "old_api.h" for the same reasons that name is used in src/mem/. It is the literal name/term for what these symbols are. These bits are _not_ gadgets of Squids latest Time accessing API (that is namespace Time and Time::Engine).

The symbols here are a mix of definitions from **several** outdated APIs that should not be used going forward. But fixing/replacing them would bloat this PR so much I left them out of scope.

Reply:
> I picked "old_api.h" for the same reasons that name is used in src/mem/. It is the literal name/term for what these symbols are.

The word "old" is not a good term to describe an API. It relays no useful information to the reader -- the API in question is as "new" as it gets to newcomers and as old as most other Squid APIs to experienced Squid developers. And those `<chrono>` APIs can be reasonably called "old" as well -- C++11 is more than 10 years old! "Old" does not imply deprecated, but many of these functions are not going away in the foreseeable future either -- we cannot deprecate the APIs used by system calls, for example. Anyway you look at it, "old" is just not a good choice here.

The (unfortunate) existence of the `src/mem/old_api.cc` file does not invalidate this reasoning. There is no need to elevate that naming mistake to the header file level.

> The symbols here are a mix of definitions from **several** outdated APIs that should not be used going forward. But fixing/replacing them would bloat this PR so much I left them out of scope.

I agree that this PR should not fix/replace those APIs. And it does not matter whether or how many of those APIs are "outdated" because, after this PR, we can and will replace or improve any outdated APIs as needed (and, ideally, without renaming the corresponding header files, but we will make those decisions when we work on those replacements).

> These bits are not gadgets of Squids latest Time accessing API (that is namespace Time and Time::Engine).

They are gadgets (for the lack of a better word). They can be declared together with future gadgets that may or may not be placed inside the Time namespace (that namespace placement decision should be made for each specific new function/type). I am not even sure we need a Time namespace, but this change request is not about that.

Reply:
Done in [d53ffe3](https://github.com/squid-cache/squid/pull/1001/commits/d53ffe3f66a8db05d3cc6327960a50594d1a80be)